extensions/fairifier-rdf-extension/dtls/fairifier/DetectFileFormatCommand.java

```diff
@@ -1,145 +1,153 @@
         String format = null;
         String url = null;
-        
-        try{
+
+        try {
             items = upload.parseRequest(req);
-            for(FileItem item:items){
-                if(item.getFieldName().equals("baseuri")){
+            for (FileItem item : items) {
+                if (item.getFieldName().equals("baseuri")) {
                     baseuri = item.getString();
-                }else if (item.getFieldName().equals("file_source")) {
+                } else if (item.getFieldName().equals("file_source")) {
                     url = item.getString();
-                }else if(item.getFieldName().equals("file_upload")){
+                } else if (item.getFieldName().equals("file_upload")) {
                     filename = item.getName();
-                    in = item.getInputStream();
+                    for (int i = 0; i < in.length; i++) {
+                        StringWriter writer = new StringWriter();
+                        in[i] = new BufferedInputStream(item.getInputStream());
+
+                    }
                 }
             }
-        }catch(FileUploadException ex){
+        } catch (FileUploadException ex) {
             respondException(res, ex);
         }
+        StringBuffer response = new StringBuffer();;
 
         if ((url != null) && !url.trim().equals("") && !url.trim().equals("url")) {
             URL obj = new URL(url);
             HttpURLConnection con = (HttpURLConnection) obj.openConnection();
             BufferedReader inr = new BufferedReader(new InputStreamReader(con.getInputStream()));
             String inputLine;
-            StringBuffer response = new StringBuffer();
 
             while ((inputLine = inr.readLine()) != null) {
-                    response.append(inputLine);
+                response.append(inputLine);
             }
             inr.close();
-            in = new ByteArrayInputStream(response.toString().getBytes());
+            for (int i = 0; i < in.length; i++) {
+                in[i] = new BufferedInputStream(
+                        new ByteArrayInputStream(response.toString().getBytes("UTF8")));
+            }
         }
-        for (RDFParser parser: parsers){
-            try{
-                if( baseuri == null) {
+        for (int i = 0; i < parsers.length; i++) {
+            try {
+                if (baseuri == null) {
                     baseuri = "";
                 }
-                parser.parse(in, baseuri);
-                format = parser.getRDFFormat().getDefaultMIMEType();
+                parsers[i].parse(in[i], baseuri);
+                format = parsers[i].getRDFFormat().getDefaultMIMEType();
                 break;
-            }catch(Exception e){
-                System.out.println(e.toString());    
+            } catch (Exception e) {
+                System.out.println(
+                        parsers[i].getRDFFormat().getDefaultMIMEType() + " : " + e.toString());
                 continue;
```

Comment:
Using exceptions as main control flow is not preferable. Try to avoid doing that.

Reply:
This is nescesarry for this method to work, i know its not prefered but the parsers airse an exception when the input cannot be parsed if this does not happen we know the file format